Diffuse/Extensions/UICollectionViewExtensions.swift

```diff
@@ -0,0 +1,24 @@
+//
+//  Copyright © FINN.no AS, Inc. All rights reserved.
+//
+
+import UIKit
+
+extension UICollectionView {
+    public func reload(with changes: CollectionChanges,
+                       section: Int = 0,
+                       updateDataSource: () -> Void) {
+        guard changes.count != 0 else { return }
+        let indexPaths = IndexPathResult(changes: changes, section: section)
+
+        performBatchUpdates({
+            updateDataSource()
```

Comment:
Does updateDataSource have to be inside batchUpdates? I don't see why there is a closure here.

Reply:
I'm not exactly sure, but I think the smartest place to update the dataSource/models is within `performBatchUpdates`. I could be wrong, seeing as I could not find any docs on this.

However, I think the main reason is thread safety. `performBatchUpdates` will run on the main thread, while the updated models could be sent to the VC on a background thread. Having `updateDataSource` within this method call will ensure everything is done on the same thread and synced.

Reply:
👍